ncstate-utility-bar.php

```diff
@@ -1,165 +1,195 @@
+        add_settings_field('ub_bar_color', 'Bar Color', array($this, 'display_color'), 'ncsu_ub_settings', 'ncsu_ub_main_section');
+        add_settings_field('ub_tf_brick', 'Use NC State Brick?', array($this, 'display_brick'), 'ncsu_ub_settings', 'ncsu_ub_main_section');
+
+    }
+
+    public function display_section()
+    {
+        ?>
+        <p>The NC State brand utility bar offers a simple way to connect our thousands of websites and let Web users
+            know where they are at all times. This element must appear at the top of all official NC State sites. No
+            content or padding of any kind may appear above this bar. The utility bar is available in various color
+            combinations (black, gray and red) consistent with official university colors, but it may not be edited or
+            altered in any way beyond the options presented when embedding. The utility bar should only be used if the
+            NC State logo is prominently displayed in the upper portion of your site.
+        </p>
+        <p>For certain entities operating under unique circumstances, the utility bar might not be required. Exceptions
+            will be handled on a case-by-case basis. Web Communications will provide support for the utility bar but
+            will rely on each unit’s internal Web or IT staff to implement the bar. If you have questions about
+            implementation or exceptions, email Web Communications at <a href="mailto:web_feedback@ncsu.edu">web_feedback@ncsu.edu</a>.
+        </p>
+
+        <p><strong>All fields are optional.</strong></p>
+        <?php
+    }
+
+    public function display_google_code()
+    {
+        ?>
+        <input type="text" size=40 name="ncsu_ub_options[ub_google_code]"
+               value="<?php echo $this->options['ub_google_code'] ?>"/>
+
+        <p>
+            <em>The search box within the utility bar can search both your own site and the entire ncsu.edu domain. To
+                search your own site you must create a search engine through <a href="https://cse.google.com/cse/"
+                                                                                target="_blank">Google Custom
+                    Search.</a> If this isn't set, the utility bar will only perform a global ncsu.edu search.</em>
+        </p>
+        <?php
+    }
+
+    public function display_placeholder()
+    {
+        echo "<input type='text' name='ncsu_ub_options[ub_search_placeholder]' value='" . preg_replace('/(\+)/', ' ', $this->options['ub_search_placeholder']) . "' />"; ?>
+        <p><em>Customization of the placeholder text in the search bar.</em></p>
+        <?php
+    }
+
+    public function display_width()
+    {
+        echo "<input type='text' name='ncsu_ub_options[ub_max_width]' value='{$this->options['ub_max_width']}' />"; ?>
+        <p><em>The utility bar is responsive. The width value should be set to your site's maximum breakpoint to enable
+                a
```

Comment:
I know this is related to wrapping long lines, but it would be nice to not have orphaned words and split tags. It would be nice to keep the link tags on a single line even if you have to wrap early.